### ISSUE TITLE
Unable to find /CLAVIN/IndexDirectory mentioned in the application.conf

### ISSUE THREAD

**Issue by s4sarath** [Berico-Technologies/CLAVIN-contrib#12]
Hi,

I have tried to run config-server from play framework. It is not able to resolve the following dir.
clavin.index = "/Users/tpinney/project/CLAVIN/IndexDirectory".

I pointed to almost all folders where I am running my project. But I am not able to find anything named or related to IndexDirectory .
![screen shot 2015-01-27 at 12 02 03 pm](https://cloud.githubusercontent.com/assets/10637096/5914494/5ee47418-a61c-11e4-97c8-48431a343c8a.png)

**Comment by charlieg** [comment]
You need to build the IndexDirectory per steps 3-6 in the [CLAVIN README](https://github.com/Berico-Technologies/CLAVIN/blob/master/README.md):

> (3.) Download the latest version of allCountries.zip gazetteer file from GeoNames.org: 
>> `curl -O http://download.geonames.org/export/dump/allCountries.zip`
>
> (4.) Unzip the GeoNames gazetteer file:
>> `unzip allCountries.zip`
>
> (5.) Compile the source code:
>> `mvn compile`
>
> (6.) Create the Lucene Index (this one-time process will take several minutes):
>> `MAVEN_OPTS="-Xmx4g" mvn exec:java -Dexec.mainClass="com.bericotech.clavin.index.IndexDirectoryBuilder"`

**Comment by charlieg** [comment]
Also, `/Users/tpinney/project/CLAVIN/IndexDirectory` is a hard-coded path. Unless your username is `tpinney`, you'll likely need to change it. :)

**Comment by s4sarath** [comment]
@charlieg - Hi Charlie , I have changed my username. Thats a basic thing. But I wasnt aware of update my indexdirectory by making use of Lucene and now everything seems good. And I ran into another exception as following. Can you guide me of how to solve it .

play - Cannot invoke the action, eventually got an error: org.apache.lucene.index.IndexFormatTooNewException: Format version is not supported (resource: NIOFSIndexInput(path="/Users/sarathrnair/Documents/Playframe/CLAVIN-contrib-master/CLAVIN-master/IndexDirectory/segments.gen")): -3 (needs to be between -2 and -2)

The format version is not supported. I am not getting what it is exactly. I am using play 2.2.1.

**Comment by charlieg** [comment]
My suspicion is you built the IndexDirectory with the latest version of CLAVIN (v2.0.0) which uses Lucene v4.9.0, while the clavin-server in the CLAVIN-contrib repository is based on an older version of CLAVIN (I'd guess v1.0.0rc7 based on the contents of [clavin-server/project/Build.scala](https://github.com/Berico-Technologies/CLAVIN-contrib/blob/master/clavin-server/project/Build.scala)).

My recommendation would be to grab the older version of CLAVIN (either checkout [this commit](https://github.com/Berico-Technologies/CLAVIN/tree/f84e86b502a69f6f3e3f67ad1e0c2281a34e07c1) or get it from [Maven Central](http://search.maven.org/#artifactdetails%7Ccom.bericotech%7Cclavin%7C1.0.0rc7%7Cjar)) and rebuild the IndexDirectory using that version of CLAVIN. The IndexDirectory it produces should hopefully be compatible with the version of CLAVIN used by clavin-server in CLAVIN-contrib.

@tlpinney might be able to add some additional guidance.

**Comment by s4sarath** [comment]
@charlieg - So i need to copy 1.0.0.rc7 jar file in the CLAVIN/src/main/java/com/bericotech/clavin directory ????

**Comment by tlpinney** [comment]
@s4sarath 
I recommend using the CLAVIN-rest if you want to set up a server. 
https://github.com/Berico-Technologies/CLAVIN-rest

The clavin-server contrib project will need to be updated.

**Comment by charlieg** [comment]
@s4sarath 
> So i need to copy 1.0.0.rc7 jar file in the CLAVIN/src/main/java/com/bericotech/clavin directory ????

No, if you're going to go down this road, I'd recommend putting the 1.0.0.rc7 jar file in a separate directory, downloading & unzipping the GeoNames gazetteer file into the same directory, then running the following command from the terminal to build the IndexDirectory:

`$ java -jar clavin-1.0.0rc7.jar com.bericotech.clavin.index.IndexDirectoryBuilder -Xmx2048M`

But, as @tlpinney recommended, you'd probably be better off using CLAVIN-rest instead of this outdated contrib project.

**Comment by s4sarath** [comment]
@charlieg and @tlpinney - thanks for your great suggestions. Working on it.
